### DISCUSSION TITLE
Ability to ignore rule on restart?

### DISCUSSION THREAD

**Discussion by albertsj1** [asmvik/yabai#1632] [Q&A]
Hi.  I have multiple displays and spaces and I use rules to have certain apps open on a specified space.

For example,
```bash
yabai -m rule --add app="^Firefox$" space=^5
```

This works great, but it has one annoyance.  When Yabai restarts, it moves apps that are already open to the specified space.  This is annoying because I often have multiple Firefox windows open and I move them to different displays.  For example, I'll move Firefox over to display 1 with docs open while I write code on display 3.  If Yabai restarts, either because of a crash, or I triggered a restart to update the configuration, it moves all Firefox windows to the space I defined.

Is there any way to tell Yabai when to use a rule or when not to use a rule?

Thanks

**Comment by asmvik** [comment]
I could add a property that makes a rule *only affect newly created windows*. Would that solve your problem?

It won't be able to tell the difference between a fresh yabai launch, a yabai restart due to crash or manual config update.
Presumably a fresh yabai launch would mean that you have no windows open, so it doesn't matter that the rule only moves newly created windows. A restart due to version update or config change or a crash, would then leave your already open windows on the space they are already assigned to.

**Comment by albertsj1** [comment]
Thanks for taking the time to respond and the suggestion.
Unless it's a solution that would work across restarts and crashes, I don't think adding a special rule property would be useful for this particular use case.  Actually, this was more annoying when Yabai was crashing on me fairly frequently.  I'm not sure what the cause of the crashing was, but I haven't experienced Yabai crashing for a while now.
I think I'll just go ahead and mark this issue as closed.
Thank you again.
